addons/l10n_it_edi/models/account_move.py

```diff
@@ -242,20 +242,40 @@ def _l10n_it_edi_get_line_values(self, reverse_charge_refund=False, is_downpayme
         """
         invoice_lines = []
         lines = self.invoice_line_ids.filtered(lambda l: l.display_type not in ('line_note', 'line_section'))
-        for num, line in enumerate(lines):
-            sign = -1 if line.move_id.is_inbound() else 1
-            price_subtotal = (line.balance * sign) if convert_to_euros else line.price_subtotal
-            # The price_subtotal should be inverted when the line is a reverse charge refund.
+        base_lines = [
+            {
+                **invl._convert_to_tax_base_line_dict(),
+                'line_record': invl,
+            }
+            for invl in lines
+        ]
+        for num, line in enumerate(base_lines):
+            line['sequence'] = num
             if reverse_charge_refund:
-                price_subtotal = -price_subtotal
-
-            # Unit price
-            price_unit = 0
-            if line.quantity and line.discount != 100.0:
-                price_unit = price_subtotal / ((1 - (line.discount or 0.0) / 100.0) * abs(line.quantity))
+                line['price_subtotal'] = -line['price_subtotal']
+            if line['discount'] != 100.0 and line['quantity']:
+                gross_price = line['price_subtotal'] / (1 - line['discount'] / 100.0)
+                line['discount_percentage_amount'] = (gross_price - line['price_subtotal']) * (-1 if reverse_charge_refund else 1)
+                line['gross_price_subtotal'] = line['currency'].round(gross_price * (-1 if reverse_charge_refund else 1))
```

Comment:
btw, I really don't like this part, really feeling like I'm doing minus of a minus of a minus...
But it's to have the same sign as what was done before for the sign of price_unit/price_subtotal in the end